### ISSUE TITLE
track_grad_norm doesn't work when accumulate_grad_batches is greater than 1

### ISSUE THREAD

**Issue by binshengliu** [PyTorchLightning/pytorch-lightning#1923]
## 🐛 Bug

When I set `accumulate_grad_batches=2`, track_grad_norm doesn't work.

It seems to me the condition for logging gradient norm can hardly be satisfied when `accumulate_grad_batches > 1 and row_log_interval(default=10) > 1`.

Say `accumulate_grad_batches=2` and `row_log_interval=2`, `(batch_idx+1)` and `batch_idx` need to be even numbers at the same time which can never be met.

https://github.com/PyTorchLightning/pytorch-lightning/blob/868dde22232389126b5342093137dbe0ab0db2a8/pytorch_lightning/trainer/training_loop.py#L608-L616

### To Reproduce

1. Set accumulate_grad_batches=2
2. Run train.fit
3. The gradient norms are not logged

### Expected behavior

The gradient norm should be logged.

### Environment

* CUDA:
	- GPU:
	- available:         False
	- version:           10.2
* Packages:
	- numpy:             1.18.4
	- pyTorch_debug:     False
	- pyTorch_version:   1.5.0
	- pytorch-lightning: 0.7.6
	- tensorboard:       2.2.0
	- tqdm:              4.45.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         
	- python:            3.7.6
	- version:           #1 SMP Debian 4.19.118-2 (2020-04-29)

**Comment by Red-Eyed** [comment]
I have same issue. Could someone please reopen the issue?
Thanks!

**Comment by binshengliu** [comment]
If you don't mind more frequent logging, specifying `row_log_interval=max(1, accumulate_grad_batches - 1)` to `Trainer` can get passed the check and do the logging.

**Comment by kyoungrok0517** [comment]
Same here. Hope to be fixed soon.

**Comment by RylanSchaeffer** [comment]
Is this still a problem?
